src/Compiler/CompilerUtils.cpp

```diff
@@ -598,10 +604,12 @@ static int genJniJar(const mlir::OwningOpRef<ModuleOp> &module,
 }
 
 // Return 0 on success, error code on failure
-static int compileModuleToObject(const mlir::OwningOpRef<ModuleOp> &module,
-    std::string outputNameWithoutExt, std::string &objectNameWithExt) {
+static int compileModuleToObject(mlir::OwningOpRef<ModuleOp> &module,
```

Comment:
Removing `const` and freeing the object internally seems difficult to follow, at least by looking at the function name. Same for other signature changes in other functions.